pkg/collector/corechecks/cluster/orchestrator/processors/k8s_pod.go

```diff
@@ -0,0 +1,157 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package processors
+
+import (
+	"fmt"
+	"strings"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/transformers"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator/redact"
+	"github.com/DataDog/datadog-agent/pkg/security/log"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	kubetypes "github.com/DataDog/datadog-agent/third_party/kubernetes/pkg/kubelet/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// K8sPodHandlers implements the Handlers interface for Kubernetes Pods.
+type K8sPodHandlers struct{}
+
+// AfterMarshalling is a handler called after resource marshalling.
+func (h *K8sPodHandlers) AfterMarshalling(ctx *ProcessorContext, resource, resourceModel interface{}, yaml []byte) (skip bool) {
+	m := resourceModel.(*model.Pod)
+	m.Yaml = yaml
+	return
+}
+
+// BeforeCacheCheck is a handler called before cache lookup.
+func (h *K8sPodHandlers) BeforeCacheCheck(ctx *ProcessorContext, resource, resourceModel interface{}) (skip bool) {
```

Comment:
💬 : I personally prefer not to use returned types for methods that are longer like here

Reply:
I only use named types in those situations, in order to add more context:
- interfaces
- methods with a good number of return args (3+)
- methods with an uncommon signature

I thought we were in the latter case and wanted to really make it explicit that the boolean return by that method indicates if we'll skip the whole resource or not after it's called. Incidentally it really emphasizes it's not skipped by default and I find it easier to spot if skip is set to true, like in:

```golang
	tags, err := tagger.Tag(kubelet.PodUIDToTaggerEntityName(string(r.UID)), collectors.HighCardinality)
	if err != nil {
		log.Debugf("Could not retrieve tags for pod: %s", err)
		skip = true
		return
	}
```